src/nnet3/nnet-component-test.cc

```diff
@@ -54,6 +54,38 @@ void TestNnetComponentAddScale(Component *c) {
   delete c3;
 }
 
+void TestNnetComponentVectorizeUnVectorize(Component *c) {
+  UpdatableComponent *uc = dynamic_cast<UpdatableComponent*>(c);
+  if((uc==NULL) || (uc->NumParameters() == 0))
+    return;
+  UpdatableComponent *uc2 = dynamic_cast<UpdatableComponent*>(uc->Copy());
+  uc2->SetZero(false);
+  Vector<BaseFloat> params(uc2->NumParameters());
+  uc2->Vectorize(&params);
+  KALDI_ASSERT(params.Min()==0.0 && params.Sum()==0.0);
+  uc->Vectorize(&params);
+  uc2->UnVectorize(params);
+  KALDI_ASSERT(uc2->Info() == uc->Info());
+  KALDI_ASSERT(uc2->DotProduct(*uc2) == uc->DotProduct(*uc));
+  delete uc2;
```

Comment:
Probably it's to do with your own connection.  You might find it easier to
do your editing on a local machine, and only push your changes to CLSP when
you want to test something.  That's what I do when my connection is flaky.
Dan


On Thu, Nov 5, 2015 at 1:24 PM, hhadian <notifications@github.com> wrote:

> In src/nnet3/nnet-component-test.cc
> <https://github.com/kaldi-asr/kaldi/pull/307#discussion_r44048355>:
>
> > @@ -54,6 +54,38 @@ void TestNnetComponentAddScale(Component *c) {
> >    delete c3;
> >  }
> >
> > +void TestNnetComponentVectorizeUnVectorize(Component *c) {
> > +  UpdatableComponent *uc = dynamic_cast<UpdatableComponent*>(c);
> > +  if((uc==NULL) || (uc->NumParameters() == 0))
> > +    return;
> > +  UpdatableComponent *uc2 = dynamic_cast<UpdatableComponent*>(uc->Copy());
> > +  uc2->SetZero(false);
> > +  Vector<BaseFloat> params(uc2->NumParameters());
> > +  uc2->Vectorize(&params);
> > +  KALDI_ASSERT(params.Min()==0.0 && params.Sum()==0.0);
> > +  uc->Vectorize(&params);
> > +  uc2->UnVectorize(params);
> > +  KALDI_ASSERT(uc2->Info() == uc->Info());
> > +  KALDI_ASSERT(uc2->DotProduct(*uc2) == uc->DotProduct(*uc));
> > +  delete uc2;
>
> Actually I was disconnected before pushing and I couldn't connect for ~1
> hour.
> Unfortunately my connection to CLSP grid is very unreliable and slow at
> some times. I'm not sure why.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/kaldi-asr/kaldi/pull/307/files#r44048355>.
>